packages/react-devtools-shared/src/hook.js

```diff
@@ -183,7 +183,7 @@ export function installHook(target: any): DevToolsHook | null {
         const breakOnConsoleErrors =
           window.__REACT_DEVTOOLS_BREAK_ON_CONSOLE_ERRORS__ === true;
         const showInlineWarningsAndErrors =
-          window.__REACT_DEVTOOLS_SHOW_INLINE_WARNINGS_AND_ERRORS__ === true;
+          window.__REACT_DEVTOOLS_SHOW_INLINE_WARNINGS_AND_ERRORS__ !== false;
```

Comment:
Should have been on by default here, to match the frontend UI setting. As a bonus this fixes RN startup use case as well.